specification/sbi_cove.adoc

```diff
@@ -1651,6 +1677,65 @@ COVE-Guest calls cause a trap to the TSM. TSM should do any processing required
 and then must forward the ECALL to the host with `scause` set to ECALL, `a7` set
 to EID, `a6` set to FID (See <<cove-fid>>), `a0`-`a5` set to ECALL args.
 
+[#sbi_covg_promote_to_tvm]
+=== Function: COVE Guest Promote to TVM (FID #0)
+[source, C]
+-----
+struct sbiret sbi_covg_promote_to_tvm(unsigned long fdt_addr,
+                                      unsigned long lap_addr);
+-----
```

Comment:
Is this function intended to be applicable only in the context of Appendix D?
If planned as a general extension, then the extended scope of the COVEG-ABI needs to be reflected in Figure 1 (similar as depicted img_11). Alternatively, it may be worth considering defining a dedicated ABI for host VM originated COVE calls.

Reply:
This is not the case, it could be use by others

Reply:
Figure 1, figure 2, and the deployment model in our figure are specific to deployment models. IF the question is whether deployment models 1 & 2 could use a one-step conversion process, the answer is yes. If for that reason you want to show the call, the I am ok with showing it in Figure 1. But some how indicating it is optional.

Reply:
I think its fine to keep it as part of COVG - I think it is understood (but can be added in the text) that this call is expected to originate from a VM not operating as a TVM. if invoked from a TVM this ABI will return an error.

Reply:
I modified the description of this call to make it clear that this call is intended for a VM and not a TVM.

Reply:
Thanks @gdhh, @wojciechozga, @rsahita for the feedback. 
Still, looking at chapter 12, I believe it may cause confusion to readers. This chapter introduces the scope of COVG by saying that it allows TVMs to communicate with their TSM. But the first function (sbi_covg_promote_to_tvm) in the following section does not comply and even states that it must not be called by a TVM. IMHO it would help if this function was pushed to the end of the chapter and introduced with some clarifying words regarding the intended deployment model (with reference to appendix D) and confirmation of applicability to the primary deployment model in Figure 1.

Reply:
@eckhard-delfs-qualcomm, thank your for your input.
Following your suggestions, I moved the `sbi_covg_promote_to_tvm()` to the end of the COVG ABI section and modified its description to make it clear that (1) it is intended for deployment model 3, (2) it is applicable (but optional) to deployment model 1 as well, (3) it is intended for VMs not TVMs.